web/components/templates/requestsV2/requestsPageV2.tsx

```diff
@@ -253,7 +256,9 @@ const RequestsPageV2 = (props: RequestsPageV2Props) => {
     isLive
   );
 
-  const requestWithoutStream = requests.find((r) => {
+  useGetS3Bodies(requestsSkeleton); // how the data ends up in the table??
```

Comment:
This is pretty weird. We are mutating the request body themselves. 

This works because the use hook is causing the whole child component to rerender, however the data object that is passed is getting re-rendered since the React Hook manager thingy is forcing this whole component to re-render.

This is working now, but will most likely break in the future. 

This is a code-smell and is not how React should be used, everything should be stateless, and all mutations should be memoized by using hooks.

----

Rules of thumb
1. NO Mutation, unless through hooks
2. use statements should return the mutated bodies

Reply:
We are going to need to fix this before merging